konflux_scripts/deploy-iqe-cji.sh

```diff
@@ -26,13 +26,22 @@ main() {
     local iqe_test_importance="${IQE_TEST_IMPORTANCE}"
     local iqe_plugins="${IQE_PLUGINS}"
     local iqe_env="${IQE_ENV:-clowder_smoke}"
+    #iqe_env_vars="ENV_VAR1=value1,ENV_VAR2=value2" -- custom set of extra environment variables to set on IQE pod
+    local iqe_env_vars="${IQE_ENV_VARS}"
     local iqe_cji_timeout="${IQE_CJI_TIMEOUT:-10m}"
 
     local selenium_arg=""
     if [[ "$selenium" == "true" ]]; then
         selenium_arg="--selenium"
     fi
 
+    iqe_env_var_args=$(awk -v IQE_ENV_VARS="$iqe_env_vars" 'BEGIN {
+      split(IQE_ENV_VARS, iqe_env_vars, ",");
+      for (i in iqe_env_vars) {
+        printf "--env-var " iqe_env_vars[i] " "
+      }
+    }')
+
```

Comment:
this is nice, thanks for bringing back support for env vars.

The only think I'd be suggesting would be to avoid the usage of `awk` in favor of bash arrays. 

check this for reference of what I mean. Basically is to iterate over a list of items, using the different elements of an array: 

https://github.com/RedHatInsights/cicd-tools/blob/d9c2a58b29caa16afa250faed7a0d44e987b11a8/src/shared/image_builder.sh#L61-L63

Reply:
I'm personally in favor of doing arrays in awk as the splitting logic is clearer and we don't need to mess with the shell's IFS to do so. It's already available on the system and in case we drop support for bash due to some reason in the future (moving to POSIX shell, busybox or dash) to cut down on surface area of attacks, I think it makes sense this isn't written bash specific.

Reply:
Not wanting to use IFS, I was trying to make the point of using a Bash array, provided the value of the variable is a spaced separated list of values. That way you could have something like 

```
IQE_ENV_VARS=("FOO=foo" "BAR=bar")

for env_var in "${IQE_ENV_VARS[@]}"; do
  env_var_params+=('--env-var' "$env_var");
done

# output is 
--env-var FOO=foo --env-var BAR=bar
```

if the comma separated values are mandatory, then I guess that option is okay ? I rarely see `awk` as an option for simply iterating values but I do agree is more readable and preferrable than to use IFS

Reply:
If we're willing to break backwards behavior, I would agree with you. But since the requirement currently is comma delimited values (which will require changing the IFS to use bash arrays), I found this cleaner than the bash solution. I feel awk is the right tool for the job here, if you disagree and we want to make a backwards breaking change I'll move this over to using bash instead.

TLDR; `awk` is already there and great for this, let's use it :)

Reply:
since we have to keep the `,` as a separator, I believe its a fair use :) , prefer this to use the IFS